scipy/stats/tests/test_censored_data.py

```diff
@@ -0,0 +1,110 @@
+        # left- or right-censored data.  Test the conversion of such
+        # an example to the canonical form in which the different
+        # types have been split into the separate arrays.
+        intervals = np.array([[0, 1],        # interval-censored
+                              [2, 2],        # not censored
+                              [3, 3],        # not censored
+                              [9, np.inf],   # right-censored
+                              [8, np.inf],   # right-censored
+                              [-np.inf, 0],  # left-censored
+                              [1, 2]])       # interval-censored
+        data = CensoredData(intervals=intervals)
+        assert_equal(data._x, [2, 3])
+        assert_equal(data._left, [0])
+        assert_equal(data._right, [9, 8])
+        assert_equal(data._intervals, [[0, 1], [1, 2]])
+
+    def test_invalid_constructor_args(self):
+        with pytest.raises(ValueError, match='must be a one-dimensional'):
+            CensoredData(x=[[1, 2, 3]])
+        with pytest.raises(ValueError, match='must be a one-dimensional'):
+            CensoredData(left=[[1, 2, 3]])
+        with pytest.raises(ValueError, match='must be a one-dimensional'):
+            CensoredData(right=[[1, 2, 3]])
+        with pytest.raises(ValueError, match='must be a two-dimensional'):
+            CensoredData(intervals=[[1, 2, 3]])
+
+        with pytest.raises(ValueError, match='must not contain nan'):
+            CensoredData(x=[1, np.nan, 2])
+        with pytest.raises(ValueError, match='must not contain nan'):
+            CensoredData(left=[1, np.nan, 2])
+        with pytest.raises(ValueError, match='must not contain nan'):
+            CensoredData(right=[1, np.nan, 2])
+        with pytest.raises(ValueError, match='must not contain nan'):
+            CensoredData(intervals=[[1, np.nan], [2, 3]])
+
+        with pytest.raises(ValueError,
+                           match='both values must not be infinite'):
+            CensoredData(intervals=[[1, 3], [2, 9], [np.inf, np.inf]])
+
+        with pytest.raises(ValueError,
+                           match='left value must not exceed the right'):
+            CensoredData(intervals=[[1, 0], [2, 2]])
+
+    @pytest.mark.parametrize('func', [CensoredData.left_censored,
+                                      CensoredData.right_censored])
+    def test_invalid_left_censored_args(self, func):
+        with pytest.raises(ValueError, match='x must be one-dimensional'):
+            func([[1, 2, 3]], [0, 1, 1])
+        with pytest.raises(ValueError,
+                           match='censored must be one-dimensional'):
+            func([1, 2, 3], [[0, 1, 1]])
+        with pytest.raises(ValueError, match='x must not contain'):
+            func([1, 2, np.nan], [0, 1, 1])
+        with pytest.raises(ValueError, match='must have the same length'):
+            func([1, 2, 3], [0, 0, 1, 1])
+
+    def test_count_censored(self):
```

Comment:
A more complete test of the `__str__` and `__repr__` methods would be good, but I won't hold this up waiting for them.